### ISSUE TITLE
Duplicate item when searching

### ISSUE THREAD

**Issue by florent1933** [vaadin/vaadin-combo-box#426]
### Description
Duplicate item when searching

### Expected outcome
Should display only one item

<img width="320" alt="capture d ecran 2017-03-27 a 00 58 46" src="https://cloud.githubusercontent.com/assets/3958123/24336024/9212447a-1288-11e7-96ac-fc7a521639a3.png">

### Actual outcome
<img width="318" alt="capture d ecran 2017-03-27 a 00 43 54" src="https://cloud.githubusercontent.com/assets/3958123/24335960/2bc19b54-1287-11e7-9e61-11b7c146009d.png">

### Steps to reproduce
1. Put a `vaadin-combo-box` element in the page

```
        <vaadin-combo-box #searchSport no-label-float
        *ngIf="activitiesTypes"
          placeholder="Activité"
          [items]="activitiesTypes"
          item-label-path="name"
          item-value-path="$key"
          [value]="activitiesTypeUid"
          (selected-item-changed)="activitiesType=$event.detail.value">
        </vaadin-combo-box>
```
with 
```
    this.activitiesTypes = [
      {
        $key : 'tennis',
        name: 'tennis'
      },
      {
        $key : 'canoe',
        name: 'canoe'
      }
    ];
```

2. Open the page in a web browser.
3. Click the input of `vaadin-combo-box` element and search an item.

### Browsers Affected
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE 11
- [ ] iOS Safari
- [x] Android Chrome

It seems to be only in in Chrome and android Chrome. I tested on firefox, safari and it's working.

It was working with the release v1.2.0 and the bug started in v1.3.0

**Comment by Saulis** [comment]
Hi!

I can't reproduce the issue in a pure Polymer app, so I'm guessing it has something to do with Angular. See https://jsfiddle.net/Saulis/6kooy6d8/

Are you getting any errors in the console?

**Comment by florent1933** [comment]
Your jsfiddle is working on my browser. 

So, you are right, it's because of angular. Do you have a template with angular and polymer for jsfiddle to reproduce this bug?

No errors in the console.

**Comment by florent1933** [comment]
I created a plunker with angular and polymer but there is no bug : https://plnkr.co/edit/pGqa0MXftgpMCSaAXCIO?p=preview

I will continue to investigate to find the reason of the bug.

**Comment by flyweight** [comment]
I'm also experiencing this using Polymer 1.8.1 and Chrome.  Doesn't seem to be an issue in other browsers.  I am starting to believe that there is a race condition in Chrome for larger apps that is causing this to happen.

**Comment by Saulis** [comment]
@flyweight are you able to produce a jsfiddle where the issue occurs?

**Comment by flyweight** [comment]
I'm trying to investigate as well where this could be happening but am getting the same results as @florent1933

**Comment by flyweight** [comment]
Just to note @Saulis: if I use your template for the item [shown here](https://gist.github.com/Saulis/48229751ba472e949b75eedc3d84de5b), then I don't see the issue.  It only appears for me if I don't include a custom template.

**Comment by Andres0101** [comment]
I'm having the same issue as @florent1933 and I think I know what the problem is. I'm using a Polymer CLI template which creates some Polymer configuration options in the index.html file. By default it looks like this:

```javascript
window.Polymer = {
  dom: 'shadow',
  lazyRegister: true,
};
```

The DOM property is set to 'shadow' by default, but if you change it to 'shady' the problem is solved. The element does not duplicate items when searching. Is there any way to make it work using 'shadow' or is the element supposed to work with 'shady' only?

**Comment by flyweight** [comment]
Confirming what @Andres0101 has stated above, changing that property resolves the issue for me as well.

**Comment by florent1933** [comment]
@Andres0101 : good job for finding the problem of this issue 👍 

It would be great if combobox can work with shadow dom because I can't use shady dom.

**Comment by Saulis** [comment]
Thanks for the help guys, here's a snippet to reproduce the issue: 
https://jsfiddle.net/Saulis/we90kg2t/

**Comment by LM450N** [comment]
I don't know if you have seen but when we search it's not possible to select a duplicate item using keyboard down arrow. Wanted precise this point.

**Comment by florent1933** [comment]
@LM450N : You are right! Amazing news!

**Comment by Saulis** [comment]
Thanks for @LM450N for spotting 4f02971!

I'm going to stamp this issue closed in `v2.0.0-alpha1` and recommend moving to use that.

For those using `1.3.x`, I've cherry-picked the fix to branch `1.x` – but beware, it has some breaking changes in that branch regarding using custom icons – they use `slot` attributes instead of classes.
